imports/ui/App.jsx

```diff
@@ -3,24 +3,43 @@ import { createContainer } from 'meteor/react-meteor-data';
 import SearchBar from './search-bar.jsx';
 import MyMap from './my-map.jsx';
 
+availableBuildings = new Mongo.Collection('available-buildings');
+
+
 // App component - represents the whole app
 class App extends Component {
+
     constructor(props){
         super(props);
         this.state = {
-            userLocation: null
+            userLocation: null,
+            buildings : [],
+            subscription: null,
         };
         this.handleMapClick = this.handleMapClick.bind(this);
     }
 
     handleMapClick(event) {
+        if (this.state.subscription!=null) {
+            this.state.subscription.stop();
+        }
         this.setState({
-            userLocation: event.latLng
+            userLocation: event.latLng,
+            subscription: Meteor.subscribe(
+                "buildings",
+                event.latLng.lat(),
+                event.latLng.lng(),
+                () => this.setState({
+                    buildings:(typeof availableBuildings.find({}).fetch()[0] !== "undefined")?
+                        availableBuildings.find({}).fetch()[0].buildings : [],
+                }),
+            ),
         });
     }
 
-    getUserLoc(){
-        if (this.state.userLocation){
+
+    getUserLocation(){
+        if (this.state.userLocation) {
             return "Lat: ".concat(this.state.userLocation.lat(), " Lng: ",this.state.userLocation.lng());
```

Comment:
Better to use the + operator or  for interpolation. This is confirmed by airbnb.